lib/plugins/aws/provider/awsProvider.js

```diff
@@ -348,39 +365,51 @@ class AwsProvider {
     const stageUpper = this.getStage() ? this.getStage().toUpperCase() : null;
 
     // add specified credentials, overriding with more specific declarations
-    try {
-      impl.addProfileCredentials(result, 'default');
-    } catch (err) {
-      if (err.message !== 'Profile default does not exist') {
-        throw err;
-      }
-    }
-    impl.addCredentials(result, this.serverless.service.provider.credentials); // config creds
-    if (this.serverless.service.provider.profile && !this.options['aws-profile']) {
-      // config profile
-      impl.addProfileCredentials(result, this.serverless.service.provider.profile);
-    }
-    impl.addEnvironmentCredentials(result, 'AWS'); // creds for all stages
-    impl.addEnvironmentProfile(result, 'AWS');
-    impl.addEnvironmentCredentials(result, `AWS_${stageUpper}`); // stage specific creds
-    impl.addEnvironmentProfile(result, `AWS_${stageUpper}`);
-    if (this.options['aws-profile']) {
-      impl.addProfileCredentials(result, this.options['aws-profile']); // CLI option profile
-    }
-    result.region = this.getRegion();
-
-    const deploymentBucketObject = this.serverless.service.provider.deploymentBucketObject;
-    if (
-      deploymentBucketObject &&
-      deploymentBucketObject.serverSideEncryption &&
-      deploymentBucketObject.serverSideEncryption === 'aws:kms'
-    ) {
-      result.signatureVersion = 'v4';
-    }
+    return impl
```

Comment:
This function was converted from sync to async, and now the memoization (in form of `this.cachedCredentials` handling) is broken (as this value is now set in async way at not certain point in a future)

Ideally if we convert it, so the whole method is memoized, and abort handling of `this.cachedCredentials`,

Probably easiest way, is via configuring following in a constructor:

```javascript
this.getCredentials = _.memoize(this.getCredentials.bind(this))
```